src/charm.py

```diff
@@ -3,162 +3,272 @@
+            raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
+        self.model.unit.status = MaintenanceStatus("K8S resources created")
+
+    def _update_profiles_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+        if not self.profiles_container.can_connect():
+            raise ErrorWithStatus("Waiting for pod startup to complete", MaintenanceStatus)
+
+        current_layer = self.profiles_container.get_plan()
+
+        if current_layer.services != self._profiles_pebble_layer.services:
+            with open("src/files/namespace-labels.yaml", encoding="utf-8") as labels_file:
+                labels = labels_file.read()
+                self.profiles_container.push(
+                    "/etc/profile-controller/namespace-labels.yaml",
+                    labels,
+                    make_dirs=True,
+                )
+            self.profiles_container.add_layer(
+                self._profiles_container_name, self._profiles_pebble_layer, combine=True
+            )
+            try:
+                self.log.info("Pebble plan updated with new configuration, replanning")
+                self.profiles_container.replan()
+            except ChangeError:
+                self.log.error(traceback.format_exc())
+                raise ErrorWithStatus("Failed to replan", BlockedStatus)
+
+    def _update_profiles_container(self, event) -> None:
+        """Updates the Profiles Pebble configuration layer if changed."""
+        if not self.profiles_container.can_connect():
+            self.unit.status = WaitingStatus("Waiting to connect to Profiles container")
+            event.defer()
+            return
+
+        try:
+            self._check_leader()
+            self.unit.status = MaintenanceStatus("Configuring Profiles layer")
+            self._update_profiles_layer()
+        except ErrorWithStatus as error:
+            raise error
+        self.unit.status = MaintenanceStatus("Profiles layer configured")
+
+    def _update_kfam_container(self, event) -> None:
+        """Updates the kfam Pebble configuration layer if changed."""
+        if not self.profiles_container.can_connect():
+            self.unit.status = WaitingStatus("Waiting to connect to kfam container")
+            event.defer()
+            return
+
+        try:
+            self._check_leader()
+            self.unit.status = MaintenanceStatus("Configuring kfam layer")
+            update_layer(
+                self._kfam_container_name,
+                self.kfam_container,
+                self._kfam_pebble_layer,
+                self.log,
+            )
+        except ErrorWithStatus as error:
+            raise error
+        self.unit.status = MaintenanceStatus("kfam layer configured")
+
+    def _on_kubeflow_profiles_ready(self, event):
+        """Define and start a workload using the Pebble API.
+        Learn more about Pebble layers at https://github.com/canonical/pebble
```

Comment:
We should change this doc string to something that describe of what is done on in this method.